### ISSUE TITLE
Build complaining about package.json 

### ISSUE THREAD

**Issue by JGSolutions** [Polymer/polymer-cli#233]
@FredKSchott 
Steps to be reproduce. On a mac with node v4.4.5

1. npm install -g polymer-cli
2. mkdir my-app
3. cd my-app
4. polymer init app-drawer-template
5. polymer build --verbose

Results:
```
error [Error: Could not find dependencies. Do you have a package.json file in your project?]
Error: Could not find dependencies. Do you have a package.json file in your project?
    at module.exports (/Users/jerrygag/node_modules/gulp-load-plugins/index.js:56:11)
    at Object.<anonymous> (/Users/jerrygag/Desktop/gulpfile.js:15:37)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at getGulpfile (/Users/jerrygag/node_modules/polymer-cli/lib/build/build.js:172:28)
    at /Users/jerrygag/node_modules/polymer-cli/lib/build/build.js:87:24
```

```
L-2225:test jerrygag$ polymer build --verbose
debug:   got args:
{ args: [ 'build', '--verbose' ] }
debug:   adding command build
debug:   adding command help
debug:   adding command init
debug:   adding command lint
debug:   adding command serve
debug:   adding command test
debug:   running...
debug:   command parsed
{ name: 'build',
  options:
   { _all: { 'sw-precache-config': 'sw-precache-config.js', verbose: true },
     global: { verbose: true },
     _none: { 'sw-precache-config': 'sw-precache-config.js' } } }
debug:   config read
{ root: '/Users/jerrygag/Desktop/esl.projects/restaurant/test',
  entrypoint: '/Users/jerrygag/Desktop/esl.projects/restaurant/test/index.html',
  shell: '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-app.html',
  fragments:
   [ '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view1.html',
     '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view2.html',
     '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view3.html' ],
  inputs:
   [ '/Users/jerrygag/Desktop/esl.projects/restaurant/test/index.html',
     '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-app.html',
     '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view1.html',
     '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view2.html',
     '/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view3.html' ] }
debug:   Running command...
debug:   building with options
{ sources: undefined,
  swPrecacheConfig: 'sw-precache-config.js',
  insertDependencyLinks: undefined,
  html: {},
  css: {},
  js: {} }
debug:   building via standard build()...
debug:   shell: /Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-app.html
debug:   entrypoint: /Users/jerrygag/Desktop/esl.projects/restaurant/test/index.html
debug:   dependencies: /Users/jerrygag/Desktop/esl.projects/restaurant/test/bower_components/**/*
debug:   sources: /Users/jerrygag/Desktop/esl.projects/restaurant/test/index.html,/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-app.html,/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view1.html,/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view2.html,/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view3.html,/Users/jerrygag/Desktop/esl.projects/restaurant/test/**/*,!build,!build/**/*,!bower_components,!bower_components/**/*,!node_modules,!node_modules/**/*
debug:   fragments: /Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view1.html,/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view2.html,/Users/jerrygag/Desktop/esl.projects/restaurant/test/src/my-view3.html
error [Error: Could not find dependencies. Do you have a package.json file in your project?]
Error: Could not find dependencies. Do you have a package.json file in your project?
    at module.exports (/Users/jerrygag/node_modules/gulp-load-plugins/index.js:56:11)
    at Object.<anonymous> (/Users/jerrygag/Desktop/gulpfile.js:15:37)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at getGulpfile (/Users/jerrygag/node_modules/polymer-cli/lib/build/build.js:172:28)
    at /Users/jerrygag/node_modules/polymer-cli/lib/build/build.js:87:24
```

**Comment by FredKSchott** [comment]
Right now we step up your directory to find and then load a gulpfile. It looks like you have one in `/Users/jerrygag/Desktop` which then tries to load plugins that it cannot find. Do you know if this gulp file throws the same error when run directly?

The advise right now would be to not run build in a project that has a gulpfile nested in a directory above it. Alternatively, you could create a blank gulpfile in your project that will be found instead of ``/Users/jerrygag/Desktop` and loaded successfully.

**Comment by FredKSchott** [comment]
@justinfagnani this seems like pretty unexpected behavior, what's the use case for searching directories above the cwd or root?

**Comment by JGSolutions** [comment]
Oh yes I do have the gulpfile which I have no clue why it's there. I think I copied it there because I needed it for reference. It's been awhile.

I removed it and the build works!

Thanks

**Comment by justinfagnani** [comment]
@FredKSchott I think looking outside of the project directory is crazy, and we shouldn't do it. But... the utilities that npm and gulp themselves use do this, and so we get that for free, fun! I think when we search for a gulpfile, we could stop if we find a package.json. This will still do fun things in some situations, but they should be much more rare.

**Comment by FredKSchott** [comment]
Fixed by #310
